build.sh

```diff
@@ -6,32 +6,94 @@
 # Copyright (C) 2018 Olliver Schinagl <oliver@schinagl.nl>
 #
 
-"${CROSS_COMPILE:=}"
-
 set -eu
 
-BUILDDIR=".build_${ARCH}"
+ARCH="${ARCH:-armhf}"
+CROSS_COMPILE="${CROSS_COMPILE:-}"
 CWD="$(pwd)"
 
-for debpkg in "*${ARCH}*".deb; do
-	if [ "${debpkg}" = "*${ARCH}*.deb" ]; then
-		continue
-	fi
-	pkglink="$(readlink "${debpkg}")"
-	if [ -L "${pkglink}" ]; then
-		echo "Removing ${pkglink}"
-		rm "${pkglink}"
-	fi
-	if [ -e "${debpkg}" ]; then
-		echo "Removing ${debpkg}"
-		rm "${debpkg}"
-	fi
+BUILD_DIR="${CWD}/.build_${ARCH}"
+
+cleanup=true
+
+cleanup()
+{
+    if ! ${cleanup}; then
+        return
+    fi
+
+    echo "Cleaning up '${BUILD_DIR}'."
+    if [ -d "${BUILD_DIR}" ] && [ -z "${BUILD_DIR##*/.build*}" ]; then
+        rm -rf "${BUILD_DIR}"
+    fi
+
+}
+
+build_prepare()
+{
+    echo "Preparing build in '${BUILD_DIR}'."
+    if [ ! -e "${BUILD_DIR}" ]; then
+        mkdir -p "${BUILD_DIR}"
+    fi
+}
+
+build()
+{
+    echo "Building software in '${BUILD_DIR}'."
+    cmake \
+        -DCMAKE_C_COMPILER="${CROSS_COMPILE}gcc" \
+        -DARCH="${ARCH}" \
+        -H"." \
```

Comment:
shouldn't there be spaces between argument and value

Reply:
according to the documentaion and examples no, they don't even quote the strings, but i found -H. to be very subtle to read. I think it even refused to work with the space! but i'm happy to retry now that it works (as it was not working before)